clientconn.go

```diff
@@ -214,9 +214,14 @@ func WithUserAgent(s string) DialOption {
 	}
 }
 
-// Dial creates a client connection the given target.
+// Dial creates a client connection to the given target.
 func Dial(target string, opts ...DialOption) (*ClientConn, error) {
-	ctx := context.Background()
+	return DialWithContext(context.Background(), target, opts...)
+}
+
+// DialWithContext creates a client connection to the given target
+// using the supplied context.
+func DialWithContext(ctx context.Context, target string, opts ...DialOption) (*ClientConn, error) {
```

Comment:
Perhaps the name should be DialContext, following the standard library's example https://golang.org/pkg/net/#Dialer.DialContext